// [[Reason: checker.typeToTypeNode deliberately fails on types that originate from node_modules.]] ////

//// [tests/cases/compiler/declarationEmitObjectAssignedDefaultExport.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,8 +1,16 @@
 
+
+//// [index.d.ts]
+import { DefaultTheme, StyledComponent } from "styled-components";
+export declare const C: StyledComponent<"div", DefaultTheme, {}, never>;
+declare const _default: invalid;
+export default _default;
+//# sourceMappingURL=index.d.ts.map
 /// [Errors] ////
 
 index.ts(7,1): error TS2742: The inferred type of 'default' cannot be named without a reference to 'styled-components/node_modules/hoist-non-react-statics'. This is likely not portable. A type annotation is necessary.
+index.ts(7,16): error TS9038: Default exports can't be inferred with --isolatedDeclarations.
 
 
 ==== node_modules/styled-components/node_modules/hoist-non-react-statics/index.d.ts (0 errors) ====
     interface Statics {
@@ -30,21 +38,27 @@
     }
     
     declare const styled: StyledInterface;
     export default styled;
-==== index.ts (1 errors) ====
+==== index.ts (2 errors) ====
     import styled, { DefaultTheme, StyledComponent } from "styled-components";
     
     const A = styled.div``;
     const B = styled.div``;
     export const C: StyledComponent<"div", DefaultTheme, {}, never> = styled.div``;
     
     export default Object.assign(A, {
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                   ~~~~~~~~~~~~~~~~~~
         B,
     ~~~~~~
+    ~~~~~~
         C
     ~~~~~
+    ~~~~~
     });
     ~~~
 !!! error TS2742: The inferred type of 'default' cannot be named without a reference to 'styled-components/node_modules/hoist-non-react-statics'. This is likely not portable. A type annotation is necessary.
+    ~~
+!!! error TS9038: Default exports can't be inferred with --isolatedDeclarations.
+!!! related TS9037 index.ts:7:1: Move the expression in default export to a variable and add a type annotation to it.
     
\ No newline at end of file
